pkg/controller/nodes/task/k8s/utils.go

```diff
@@ -0,0 +1,35 @@
+package k8s
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/flyteorg/flytepropeller/pkg/utils"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+var dns1123InvalidRegex = regexp.MustCompile("[^-a-z0-9]")
+var camelCaseRegex = regexp.MustCompile("([a-z0-9])([A-Z])")
+
+const maxUniqueIDLength = 20
+
+func ConvertToDNS1123CompatibleString(name string) string {
+	name = ConvertCamelCaseToKebabCase(name) // best effort to preserve readability for Java class name
+	name = strings.ToLower(name)
+	name = dns1123InvalidRegex.ReplaceAllString(name, "")
+	name = strings.Trim(name, ".-")
+	if len(name) > validation.DNS1123LabelMaxLength {
+		fixedLengthID, err := utils.FixedLengthUniqueID(name, maxUniqueIDLength)
+		if err == nil {
+			name = name[:validation.DNS1123LabelMaxLength-maxUniqueIDLength-1] + "-" + fixedLengthID
+		} else {
+			name = name[:validation.DNS1123LabelMaxLength]
+		}
+	}
+	return name
+}
+
+func ConvertCamelCaseToKebabCase(name string) string {
```

Comment:
The other reason to keep these APIs in pluginsmachinary as you initially did is to be able to use them here:
https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/plugins/array/k8s/task.go#L115

Which does not use PluginManager to kick off pods... and will need the same handling... I believe

Reply:
I moved them here because of `encoder.go`. Do you want `encoder.go` moved to flyteplugins as well?

Reply:
ic... yeah sure... maybe actually make a better package name in pluginmachinary for this? `encoding` or something?